cmd/titus-inject-metadataproxy/main.go

```diff
@@ -0,0 +1,61 @@
+package main
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/Netflix/titus-executor/logger"
+	"github.com/Netflix/titus-executor/metadataserver/inject"
+	log2 "github.com/Netflix/titus-executor/utils/log"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	pkgviper "github.com/spf13/viper"
+)
+
+func main() {
+	log2.MaybeSetupLoggerIfOnJournaldAvailable()
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	logruslogger := logrus.New()
+	ctx = logger.WithLogger(ctx, logruslogger)
+
+	v := pkgviper.New()
+	rootCmd := &cobra.Command{
+		Args: cobra.ArbitraryArgs,
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			if err := v.BindPFlags(cmd.Flags()); err != nil {
+				panic(err)
+			}
+
+			switch level := v.GetString("log-level"); level {
+			case "info":
+				logruslogger.SetLevel(logrus.InfoLevel)
+
+			case "debug":
+				logruslogger.SetLevel(logrus.DebugLevel)
+			default:
+				return fmt.Errorf("Cannot parse log-level: %q", level)
+			}
+
+			return nil
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return inject.Inject(ctx, v.GetString("titus-pid-1-dir"), args)
+		},
+	}
+	rootCmd.Flags().String("titus-pid-1-dir", "", "Listening address")
```

Comment:
While it is nice that we have these two commands (the injector and the imds service itself) and it is nice to separate the concerns, now that they are in golang you *could* just have one command.

We would only need to do that if you thought it would actually simplify things, and not make things more complex. But lgtm like this.

Reply:
We can't, because of threads we need to re-exec in order to complete the namespace transition.